### DISCUSSION TITLE
How to extend search radius to find the nearest edge in the network?

### DISCUSSION THREAD

**Discussion by tobwen** [valhalla/valhalla#5012] [Q&A]
### problem
I'm trying to find a route from `49.235084, 8.093549` to `49.197403, 8.111968` (lat, lon):
https://valhalla.openstreetmap.de/directions?profile=car&wps=8.093549,49.235084,8.111968,49.197403

Since the target is in a pedestrial zone, it can't directly route of course. 

### question
Can I raise the search radius to find the nearest edge in the network?

### what I tried
- shortest=true
- radius=5000
- node_snap_tolerance=5000
- street_side_tolerance=5000
- street_side_max_distance=5000
- preferred_side=either

And in the config true/false in every combination:
- "costmatrix_allow_second_pass": false
- "costmatrix_check_reverse_connection": false

### update
Wow... For this, Valhalla finds a route through the pedestrial area, but doesn't use the shortest one (`costing=car`):

from: `49.198262, 8.110933`, to: `49.198243, 8.111872`

https://valhalla.openstreetmap.de/directions?profile=car&wps=8.110933,49.198262,8.111872,49.198243

**Comment by chrstnbwnkl** [comment]
Usually `radius` should do the trick but it's possible the server side config clamps your value of 5000 to something smaller in order not to blow up during the candidate search (default is 200m).

**Reply by tobwen** [reply]
> Usually `radius` should do the trick but it's possible the server side config clamps your value of 5000 to something smaller in order not to blow up during the candidate search (default is 200m).

I'm using the container default config, which has radius set to `0` (the docs say, `0` is default):

``` json
        "service_defaults": {
          "heading_tolerance": 60,
          "minimum_reachability": 50,
          "node_snap_tolerance": 5,
          "radius": 0,
          "search_cutoff": 35000,
          "street_side_max_distance": 1000,
          "street_side_tolerance": 5
        },
```

When I set it as `service_defaults`, it works! But it dramatically (!) slows down ALL the requests. Setting it as a query parameter doesn't seem to work here. Neither to `/directions`, nor to `/sources_to_targets`.

Maybe the parameter isn't sent to the backend or the backend "wins" all the time?

**Reply by chrstnbwnkl** [reply]
Sorry, I mean the `max_radius` parameter. If you set that one higher, you should be able to control the radius properly. If you just set the default value to something that high, I'd expect requests in urban areas to be significantly slower.

**Reply by tobwen** [reply]
>  I mean the `max_radius` parameter. 

Ok. I've set `"max_radius": 2000,` in the config (and restarted), but `radius` still gets ignored:

```
echo '{"id" : "100mN29042E41821", "locations" : [{"lat" : 49.23509664, "lon" : 8.09363118}, {"lat" : 49.197853, "lon" : 8.11214}]}' \
| xargs -d $'\n' -P10 -i curl -skL 'http://localhost:8002/route?costing=auto&radius=5000' -d '{}'; echo

{"error_code":442,"error":"No path could be found for input","status_code":400,"status":"Bad Request"}
```

Works perfectly with `costing=pedestrian` (just to check than anything else is fine).

**Reply by chrstnbwnkl** [reply]
You need to set `radius` for each location.
